File: sprint6/By languages/C#/lesson5code1.cs
using System;
using System.Collections.Generic;

List<string> color = new List<string> {"white", "white", ...};
int time = 0;
List<int?> entry = new List<int?> {null, null, ...};
List<int?> leave = new List<int?> {null, null, ...};

void DFS(int v) {
    time += 1;
    entry[v] = time;
    color[v] = "gray";
    foreach (int w in getOutgoingEdges(v)) {
        if (color[w] == "white") {
            DFS(w);
        }
    }
    time += 1;
    leave[v] = time;
    color[v] = "black";
}

File: sprint6/By languages/swift/lesson5code1.swift
import Foundation

enum Color {
    case white
    case gray
    case black
}

// Замените V на нужное значение
var color: [Color] = Array(repeating: .white, count: V) 
var time: Int = 0
var entry: [Int] = Array(repeating: 0, count: V)  
var leave: [Int] = Array(repeating: 0, count: V)  

func DFS(_ v: Int, getOutgoingEdges: (Int) -> [Int]) {
    time += 1
    entry[v] = time
    color[v] = .gray
    
    for w in getOutgoingEdges(v) {
        if color[w] == .white {
            DFS(w, getOutgoingEdges: getOutgoingEdges)
        }
    }
    
    time += 1
    leave[v] = time
    color[v] = .black
}
